vendor/bat-native-ads/src/bat/ads/internal/ads/new_tab_page_ads/new_tab_page_ad.cc

```diff
@@ -44,13 +45,11 @@ void NewTabPageAd::FireEvent(const std::string& uuid,
     return;
   }
 
-  // TODO(https://github.com/brave/brave-browser/issues/14015): Refactor this
-  // logic to only apply frequency capping if the new tab page ad was not served
-  // by the library. |AdServing::MaybeServeAd| is responsible for applying
-  // frequency caps for new tab page ads served by the library
+  // Apply permission rules frequency capping for new tab page ad view events
+  // if Brave Ads are disabled.
   new_tab_page_ads::frequency_capping::PermissionRules permission_rules;
   if (event_type == mojom::NewTabPageAdEventType::kViewed &&
-      !permission_rules.HasPermission()) {
+      !permission_rules.HasPermission() && !ShouldRewardUser()) {
```

Comment:
nit: we should evaluate `ShouldRewardUser()` before `HasPermission()` to lower performance